src/utils/textUtils.js

```diff
@@ -0,0 +1,5 @@
+function lineCounter(string) {
+  return string.split('\n').length;
```

Comment:
You could implement this as a simple for loop to check count each newline character instead of using split. That would avoid allocations and probably be faster. Want to test?

Reply:
I wrote a jsPerf for this in it appears regex is faster than both split and for loop.
https://jsperf.com/linecounter/1
I'll change the function into the regex method